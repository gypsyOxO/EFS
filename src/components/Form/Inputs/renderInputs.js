import React from "react"
import TextField from "@material-ui/core/TextField"
//import { RadioButton, RadioButtonGroup } from 'material-ui/RadioButton';
import Checkbox from "@material-ui/core/Checkbox"
import { Select } from "@material-ui/core"
import Radio from "@material-ui/core/Radio"
import RadioGroup from "@material-ui/core/RadioGroup"

export const renderTextField = ({ field, ...props
	// input,
	// label,
	// meta: { touched, error },
	// ...custom
}) => (
	<TextField
        {...field}
        {...props}
		// hintText={label}
		variant="outlined"
		// floatingLabelText={label}
		// errorText={touched && error}
		// label={label}
		// {...input}
		// {...custom}
	/>
)

export const renderCheckbox = ({ input, label }) => (
	<Checkbox
		label={label}
		checked={input.value ? true : false}
		onCheck={input.onChange}
	/>
)

export const renderRadioGroup = ({field, ...props}) => {
	return (
	<RadioGroup
		{...field}
        {...props}        		
	/>)
}

export const renderSelectField = ({
	input,
	label,
	meta: { touched, error },
	children,
	...custom
}) => (
	<Select
		floatingLabelText={label}
		variant="outlined"
		errorText={touched && error}
		{...input}
		onChange={(event, index, value) => input.onChange(value)}
		children={children}
		{...custom}
	/>
)
