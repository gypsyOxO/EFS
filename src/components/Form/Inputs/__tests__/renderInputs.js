import React from "react"
import { shallow } from "enzyme"
import { renderReactSelectField, renderDatePicker, renderTextField, renderCheckbox, renderRadioGroup } from "../renderInputs"

describe("renderInputs", () => {
	it('renders renderReactSelectField', () => {
		const component = shallow(<renderReactSelectField />)
		expect(component).toMatchSnapshot()
	})
})


describe("renderInputs", () => {
	it('renders renderDatePicker', () => {
		const component = shallow(<renderDatePicker />)
		expect(component).toMatchSnapshot()
	})
})

describe("renderInputs", () => {
	it('renders renderTextField', () => {
		const component = shallow(<renderTextField />)
		expect(component).toMatchSnapshot()
	})
})

describe("renderInputs", () => {
	it('renders renderCheckbox', () => {
		const component = shallow(<renderCheckbox />)
		expect(component).toMatchSnapshot()
	})
})

describe("renderInputs", () => {
	it('renders renderRadioGroup', () => {
		const component = shallow(<renderRadioGroup />)
		expect(component).toMatchSnapshot()
	})
})

