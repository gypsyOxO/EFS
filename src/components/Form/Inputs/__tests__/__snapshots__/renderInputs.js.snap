// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderInputs renders renderCheckbox 1`] = `<renderCheckbox />`;

exports[`renderInputs renders renderDatePicker 1`] = `<renderDatePicker />`;

exports[`renderInputs renders renderRadioGroup 1`] = `<renderRadioGroup />`;

exports[`renderInputs renders renderReactSelectField 1`] = `<renderReactSelectField />`;

exports[`renderInputs renders renderTextField 1`] = `<renderTextField />`;
